flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -209,20 +209,23 @@ public void registerTempCatalogScalarFunction(ObjectIdentifier oi, ScalarFunctio
 	private Set<String> getUserDefinedFunctionNames() {
 		Set<String> result = new HashSet<>();
 
-		// Get functions in catalog
+		// add temp system functions
+		result.addAll(tempSystemFunctions.keySet());
+
+		// add temp catalog functions
+		result.addAll(tempCatalogFunctions.keySet().stream()
```

Comment:
Shouldn't we add only functions available in the current namespace?

Reply:
right